### ISSUE TITLE
Incorrect results slicing on inner dim for RM f32 Tensors with ttnn.slice

### ISSUE THREAD

**Issue by jaykru-tt** [tenstorrent/tt-metal#12645]
**Describe the bug**
It seems like slice is ignoring the first index when slicing; e.g. if `A` is a torch tensor and `A_tt` is its ttnn Tensor equivalent, A_tt[2:4] == A[0:2]

**To Reproduce**
Run this script on 18cd074cc3b347faf2b204cbeee85cfc0748e49b:
```
import ttnn
import torch

A = torch.rand(1,1,1,4)

wh = ttnn.open_device(0)

A_tt = ttnn.from_torch(A).to(wh)

our_res = ttnn.split(A_tt,4,dim=3)
print(our_res) # all entries equal A[0]
our_res = [ttnn.from_device(x).to_torch() for x in our_res]
torch_res = torch.split(A,1,dim=3)
print(our_res)
print(torch_res)

assert(len(our_res) == len(torch_res))
assert(all(torch.equal(our_res[i],torch_res[i]) for i in range(len(our_res))))
```

**Expected behavior**
Slicing behavior should match torch.

**Comment by prajaramanTT** [comment]
@jaykru-tt Is this still an open issue ? If not, can you please close this ? Thanks.

**Comment by jaykru-tt** [comment]
@sjameelTT this is fixed now right?

**Comment by zzigler-tt** [comment]
@rfurko-tt can you please review and determine if this is still an issues or can be closed out? Thanks
